lib/ansible/module_utils/facts/system/systemd.py

```diff
@@ -42,6 +42,9 @@ def collect(self, module=None, collected_facts=None):
 
             systemd_facts["systemd"] = {}
             systemd_facts["systemd"]["features"] = str(stdout.split("\n")[1])
-            systemd_facts["systemd"]["version"] = int(stdout.split(" ")[1])
+            systemd_facts["systemd"]["full_version"] = str(
+                (stdout.split(" ")[2]).split(")")[0][1:],
+            )
+            systemd_facts["systemd"]["version"] = int(stdout.split(" ")[1][:3])
```

Comment:
@jborean93 found alternate and probably more parsable way to get version
`busctl --system get-property org.freedesktop.systemd1 /org/freedesktop/systemd1 org.freedesktop.systemd1.Manager Version`

Reply:
It looks like it might suffer from the same problems here as `Version` displays the "full_version" result and there are no other dbus properties to get the "major" version number that `systemctl --version` displays. I'm not sure what the best method would be here but potentially we should extract the values in a similar way to this PR but have a catch for the int cast and fallback to the raw string value if that fails?

Reply:
I'll write something with the catch to test it out. 

Another way is to just dump the "full version" and just have a version string with everything in it and let the user parse it in a task and the result as a fact.

Reply:
it will now return an int if type int (which is the case for the majority) or return a str if `ValueError`. 

```
127.0.0.1 | SUCCESS => {
    "ansible_facts": {
        "ansible_systemd": {
            "features": "+PAM +AUDIT +SELINUX +APPARMOR +IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBCRYPTSETUP_PLUGINS +LIBFDISK +PCRE2 -PWQUALITY +P11KIT +QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD -BPF_FRAMEWORK -XKBCOMMON +UTMP +SYSVINIT -LIBARCHIVE",
            "full_version": "256~rc2",
            "version": "256~rc2"
        },
```

```
ok: [testhost] => {
    "msg": {
        "features": "+PAM +AUDIT +SELINUX -APPARMOR +IMA +SMACK +SECCOMP -GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN -IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 +PWQUALITY +P11KIT +QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +BPF_FRAMEWORK +XKBCOMMON +UTMP +SYSVINIT default-hierarchy=unified",
        "full_version": "254.10-1.fc39",
        "version": 254
    }
}
```